src/pages/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/index.mdx

```diff
@@ -0,0 +1,1242 @@
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+## Storage.list
+
+<Accordion title='Input' headingLevel='3'>
+  <Columns columns={2}>
+    <div>
+      **V5**
+
+      ```
+      // Assumes default provider 'AWSS3': custom provider not supported in v6
+      path: string,
+      config?: {
+        level?: 'public' | 'protected' | 'private';
+        provider?: 'AWSS3';
+        customPrefix?: {
+          private?: string;
+          public?: string;
+          protected?: string;
+        };
+        track?: boolean; // Not yet supported in V6
+        bucket?: string;
+        pageSize?: number | 'ALL';
+        identityId?: string;
+        nextToken?: string;
+      }
+      ```
+
+    </div>
+    <div>
+      **V6**
+
+      ```
+      // Paginated List
+      input {
+        prefix?: string;
+        options?: {
+          listAll?: false;
+          accessLevel?: 'guest' | 'private' | 'protected';
+          pageSize?: number;
+          nextToken?: string;
+          useAccelerateEndpoint?: boolean;
+          // Included if accessLevel is 'protected'
+          targetIdentityId?: string';
+        };
+      }
+
+      // List All
+      input {
+        prefix?: string;
+        options?: {
+          listAll: true;
+          accessLevel?: 'guest' | 'private' | 'protected';
+          useAccelerateEndpoint?: boolean;
+          // Included if accessLevel is 'protected'
+          targetIdentityId?: string';
+        };
+      }
+      ```
+      
+    </div>
+  </Columns>
+</Accordion>
+
+<Accordion title='Output' headingLevel='3'>
+  <Columns columns={2}>
+    <div>
+      **V5**
+
+      ```
+      StorageListOutput {
+        results: S3ProviderListOutputItem[];
+        nextToken?: string;
+        hasNextToken: boolean;
+      };
+
+      S3ProviderListOutputItem {
+        key: string;
+        eTag: string;
+        lastModified: Date;
+        size: number;
+      }
+      ```
+
+    </div>
+    <div>
+      **V6**
+
+      ```
+      // Paginated List
+      ListPaginateOutput {
+        items: ListOutputItem[];
+        nextToken?: string;
+      }
+
+      // List All
+      ListAllOutput {
+        items: ListOutputItem[];
+      }
+
+      ListOutputItem {
+        key: string;
+        lastModified?: Date;
+        size?: number;
+        eTag?: string;
+      }
+      ```
+
+    </div>
+  </Columns>
+</Accordion>
+
+### Paginated List (Default)
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { list } from 'aws-amplify/storage';
+
+    const handleList = async (
+      prefix: string, 
+      accessLevel: string,
+      id: string
+    ) => {
+      const result = Storage.list({
+        prefix,
+        options: {
+          // accessLevel and targetIdentityId not required if accessLevel is 'guest'
+          accessLevel,
+          targetIdentityId: id
+        }
+      });
+      const { items } = result;
+      const nextToken = result.nextToken || null;
+    }
+    ```
```

Comment:
This is not the best code snippet to show the pagination. I'd recommend a code snippet actually make make multiple list API calls with `nextToken`. Similar to [v5 example](https://docs.amplify.aws/javascript/prev/build-a-backend/storage/list/#paginated-file-access)

Reply:
Good idea, can do.